### ISSUE TITLE
Persistent daemon.json Configuration Across Docker/Snap Updates

### ISSUE THREAD

**Issue by busybeaver** [docker-snap/docker-snap#39]
Hi,

the issue is similar to [the comment in issue #38](https://github.com/docker-snap/docker-snap/issues/38#issuecomment-845893398):

Is it possible to add configuration to the `daemon.json` file that are persisted across docker/snap updates? Right now, the file is stored at `/var/snap/docker/current/config/daemon.json`. However, since the `current` folder only links to the latest version, updating the snap/docker package results in a "reset" configuration.

Use Case:
I add TLS validation to access to Docker daemon remotely in a secure way (templated JSON partially below):
```
    "hosts" : [ "tcp://${var.host.ip}:2376", "unix:///var/run/docker.sock" ],
    "tls": true,
    "tlscacert": linux_file.docker_host_ca_cert.path,
    "tlscert": linux_file.docker_host_server_cert.path,
    "tlskey": linux_file.docker_host_server_key.path,
    "tlsverify": true
```

Getting these changes reset/removed after each update, defeats the purpose of this configuration a bit (having to SSH into the machine and redo the change).

Summing up, the reason is that the `daemon.json` file is located in the `$SNAP_DATA` folder (instead of the `$SNAP_COMMON` folder), which gets replaced during snap updates. So my question is: is it somehow possible to make the mentioned changes above persistent?

**Comment by busybeaver** [comment]
Could we add something link this below to the `dockerd-wrapper` file?

```bash
# use the docker daemon config file from the SNAP_COMMON folder
DOCKER_CONFIG_FILE=$SNAP_COMMON/config/daemon.json
if [ ! -f "$SNAP_COMMON/config/daemon.json" ]; then
    # if the file isn't present, use the daemon config file from the SNAP_DATA folder
    DOCKER_CONFIG_FILE=$SNAP_DATA/config/daemon.json
fi
#...
exec dockerd \
	--group "$default_socket_group" \
	--exec-root="$run_dir" \
	--data-root="$SNAP_COMMON/var-lib-docker" \
	--pidfile="$run_dir/docker.pid" \
       # line below changed
	--config-file="$DOCKER_CONFIG_FILE" \
	"$@"
```

Or is there a better/smarter way to handling this? Maybe even without changes needed to the package?

**Comment by anonymouse64** [comment]
The files from $SNAP_DATA are copied from the previous revision to the new revision when a refresh occurs, is that not what you are seeing? Can you install the revision from stable, then modify the daemon.json and then refresh to the revision from edge and show the two files from the previous stable revision daemon.json and the edge revision daemon.json ?

**Comment by busybeaver** [comment]
Hmmm... I was sure that this happened in the past. But now testing it with `snap refresh --edge` it seems to indeed copy the file. 
Will close the issue and reopen if it happens again. Thanks @anonymouse64
